pkg/process/config/config.go

```diff
@@ -356,24 +359,40 @@ func loadEnvVariables() {
 		"DD_LOG_LEVEL":      "log_level",
 	} {
 		if v, ok := os.LookupEnv(envKey); ok {
+			// We don't want to overwrite environment variables with encrypted secrets
+			// Note: This will cause decryption to break for non-standard environment flags used in agent 5 which may
+			//       hurt users transitioning from A5 to A6, but since secrets aren't supported in A5, it's likely
+			//       not an issue.
+			if isEncryptedSecret(v) {
+				continue
+			}
+
 			config.Datadog.Set(cfgKey, v)
 		}
 	}
 
 	// Support API_KEY and DD_API_KEY but prefer DD_API_KEY.
-	if key, ddkey := os.Getenv("API_KEY"), os.Getenv("DD_API_KEY"); ddkey != "" {
-		log.Info("overriding API key from env DD_API_KEY value")
-		config.Datadog.Set("api_key", strings.TrimSpace(strings.Split(ddkey, ",")[0]))
-	} else if key != "" {
-		log.Info("overriding API key from env API_KEY value")
-		config.Datadog.Set("api_key", strings.TrimSpace(strings.Split(key, ",")[0]))
+	apiKey, envKey := os.Getenv("DD_API_KEY"), "DD_API_KEY"
+	if apiKey == "" {
+		apiKey, envKey = os.Getenv("API_KEY"), "API_KEY"
+	}
+
+	if apiKey != "" && !isEncryptedSecret(apiKey) { // We don't want to overwrite the API KEY provided as an environment variable
+		log.Infof("overriding API key from env %s value", envKey)
+		config.Datadog.Set("api_key", strings.TrimSpace(strings.Split(apiKey, ",")[0]))
 	}
 
-	if v := os.Getenv("DD_CUSTOM_SENSITIVE_WORDS"); v != "" {
+	if v := os.Getenv("DD_CUSTOM_SENSITIVE_WORDS"); v != "" && !isEncryptedSecret(v) {
 		config.Datadog.Set("process_config.custom_sensitive_words", strings.Split(v, ","))
 	}
 }
 
+// Note: this is copied from isEnc() @ pkg/secrets/secrets.go
+func isEncryptedSecret(str string) bool {
```

Comment:
This is duplicated from [pkg/secrets](https://github.com/DataDog/datadog-agent/blob/86fa33f9b855154b0345211a120807a3797e772c/pkg/secrets/secrets.go#L110). If we ever find a bug there, and change something, nobody will know to change this too.

Reply:
Yes, this is a just a temporary measure to unblock the release. All of this code that overrides environment vars needs to be fixed.